flow/flow.go

```diff
@@ -304,23 +308,54 @@ func (p *Packet) TransportFlow() (gopacket.Flow, error) {
 // Keys returns keys of the packet
 func (p *Packet) Keys(parentUUID string, uuids *UUIDs, opts *Opts) (uint64, uint64, uint64) {
 	hasher := xxHash64.New(0)
-	if layer := p.NetworkLayer(); layer != nil {
-		Hash(layer.NetworkFlow(), hasher)
+	swap := false
+
+	l2Layer := p.LinkLayer()
+	l3Layer := p.NetworkLayer()
+	swapFromNetwork := true
```

Comment:
I would write the following block like this (more compact)

```
	swapByNetwork := true

	if (opts.LayerKeyMode == L2KeyMode && l2Layer != nil) ||
		l3Layer == nil {

		swapByNetwork = false
		src, dst := l2Layer.LinkFlow().Endpoints()
		cmp := bytes.Compare(src.Raw(), dst.Raw())
		if cmp == 0 && l3Layer != nil {
			swapByNetwork = true
		} else {
			swap = cmp > 0
		}
	}
	if swapByNetwork {
		src, dst := l3Layer.NetworkFlow().Endpoints()
		if cmp := bytes.Compare(src.Raw(), dst.Raw()); cmp == 0 {
			if tf, err := p.TransportFlow(false); err == nil {
				src, dst := tf.Endpoints()
				swap = bytes.Compare(src.Raw(), dst.Raw()) > 0
			}
		} else {
			swap = cmp > 0
		}
	}
```

Reply:
I did a mix